crates/storage-ops/src/rollback/node/batch_prover.rs

```diff
@@ -104,27 +103,21 @@ impl BatchProverLedgerRollback {
             let job_id = job_record.key;
             let commitment_indices = job_record.value;
 
-            if commitment_indices
+            if !commitment_indices
                 .iter()
-                .any(|&idx| idx > last_sequencer_commitment_index)
+                .all(|&idx| idx > last_sequencer_commitment_index)
             {
-                jobs_to_check.push((job_id, commitment_indices));
-            }
-        }
-
-        // Process collected jobs
-        for (job_id, commitment_indices) in jobs_to_check {
-            // If ANY index should be preserved, skip the whole job
-            if commitment_indices
-                .iter()
-                .any(|&idx| idx <= last_sequencer_commitment_index)
-            {
-                tracing::warn!(
-                    "Preserving job {} that spans rollback boundary. Job indices: {:?}, rollback target: {}",
-                    job_id,
-                    commitment_indices,
-                    last_sequencer_commitment_index
-                );
+                if commitment_indices
```

Comment:
I don't think this if statement is necessary. The outer one will do the job for the tracing warn

Reply:
my reasoning was that, if the rollback index is 10, but a job has 1,2,3 indices, then we don't print it as its completely out of the bounds. but if a job has 9,10,11, only then we print this as its transgressing on both sides. wdyt?

Reply:
Done https://github.com/chainwayxyz/citrea/pull/2618/commits/2b2e53ec28d14ef0c55f929833ed0753315db2d9